modules/financial_bot/financial_bot/chains.py

```diff
@@ -0,0 +1,82 @@
+from typing import Any, Dict, List
+
+import qdrant_client
+from financial_bot.embeddings import EmbeddingModelSingleton
+from financial_bot.template import PromptTemplate
+from langchain.chains.base import Chain
+from langchain.llms import HuggingFacePipeline
+
+
+class ContextExtractorChain(Chain):
+    """
+    Encode the question, search the vector store for top-k articles and return
+    context news from documents collection of Alpaca news.
+    """
+
+    top_k: int = 1
+    embedding_model: EmbeddingModelSingleton
+    vector_store: qdrant_client.QdrantClient
+    vector_collection: str
+    output_key: str = "payload"
+
+    @property
+    def input_keys(self) -> List[str]:
+        return ["about_me", "question", "context"]
+
+    @property
+    def output_keys(self) -> List[str]:
+        return [self.output_key]
+
+    def _call(self, inputs: Dict[str, Any]) -> Dict[str, Any]:
+        # TODO: handle that None, without the need to enter chain
+        about_key, quest_key, contx_key = self.input_keys
+        question_str = inputs.get(quest_key, None)
+
+        # TODO: maybe async embed?
+        embeddings = self.embedding_model(question_str)
+
+        # TODO: get rid of hardcoded collection_name, specify 1 top_k or adjust multiple context insertions
+        matches = self.vector_store.search(
+            query_vector=embeddings,
+            k=self.top_k,
+            collection_name=self.vector_collection,
+        )
+
+        content = matches[0].payload["summary"]
+
```

Comment:
What happens if we have `top_k > 1`? I was thinking of setting it to 3

Reply:
I've adapted to merge all summaries into one big string with "\n" sep between them. Currently the top-k is set from constants